MREUnityRuntime/MREUnityRuntimeLib/Patching/Types/PhysicsBridgePatch.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using MixedRealityExtension.Animation;
+using MixedRealityExtension.Core.Physics;
+using Newtonsoft.Json.Linq;
+
+namespace MixedRealityExtension.Patching.Types
+{
+	public class TransformPatchInfo
+	{
+		public TransformPatchInfo() { }
+
+		internal TransformPatchInfo(Guid id, RigidBodyTransform transform)
+		{
+			Id = id;
+
+			Transform = new TransformPatch();
+			Transform.Position = new Vector3Patch(transform.Position);
+			Transform.Rotation = new QuaternionPatch(transform.Rotation);
+		}
+
+		/// <summary>
+		/// ID of the actor (of the RB)
+		/// </summary>
+		public Guid Id { get; set; }
+
+		public TransformPatch Transform { get; set; }
+	}
+
+	public class PhysicsBridgePatch : IPatchable
+	{
+		public PhysicsBridgePatch()
+		{
+			TransformCount = 0;
+			TransformsBLOB = null;
+		}
+
+		internal PhysicsBridgePatch(Guid sourceId, Snapshot snapshot)
+		{
+			Id = sourceId;
+			Time = snapshot.Time;
+
+			TransformCount = snapshot.Transforms.Count;
+
+			if (TransformCount > 0)
+			{
+				// copy transform to native array to reinterpret it to byte array without 'unsafe' code
+				// todo: we should use native array in snapshot anyway.
+				Unity.Collections.NativeArray<Snapshot.TransformInfo> transforms =
+						new Unity.Collections.NativeArray<Snapshot.TransformInfo>(snapshot.Transforms.ToArray(), Unity.Collections.Allocator.Temp);
+
+				int sizeOfTransformInfo = Unity.Collections.LowLevel.Unsafe.UnsafeUtility.SizeOf<Snapshot.TransformInfo>();
+				TransformsBLOB = transforms.Reinterpret<byte>(sizeOfTransformInfo).ToArray();
+			}
+		}
+
+		internal Snapshot ToSnapshot()
+		{
+			if (TransformCount > 0)
+			{
+					Unity.Collections.NativeArray<byte> blob =
+						new Unity.Collections.NativeArray<byte>(TransformsBLOB, Unity.Collections.Allocator.Temp);
+
+					// todo: use native array in snapshot
+					Unity.Collections.NativeArray<Snapshot.TransformInfo> transforms = blob.Reinterpret<Snapshot.TransformInfo>(1);
+					return new Snapshot(Time, new List<Snapshot.TransformInfo>(transforms.ToArray()));
+			}
+
+			return new Snapshot(Time, new List<Snapshot.TransformInfo>());
+		}
+
+		/// <summary>
+		/// source app id (of the sender)
+		/// </summary>
+		public Guid Id { get; set; }
+
+		public float Time { get; set; }
+
+		public int TransformCount { get; set; }
+
+		public byte[] TransformsBLOB { get; set; }
```

Comment:
How does Newtonsoft.Json serialize this byte array? Base64?

Reply:
Wondering if this needs an explicit converter.

Reply:
According to https://www.newtonsoft.com/json/help/html/SerializationGuide.htm
it should be String (base 64 encoded)